07-allow-traffic-from-some-pods-in-another-namespace.md

```diff
@@ -1,23 +1,109 @@
 # ALLOW traffic from some pods in another namespace
 
-![#f03c15](https://placehold.it/15/f03c15/000000?text=+) **This policy is not possible today.** ![#f03c15](https://placehold.it/15/f03c15/000000?text=+)
+![#f03c15](https://placehold.it/15/f03c15/000000?text=+) **This policy is currently not enforced by any networking plugin. It will be enforced by Calico v.3.2** ![#f03c15](https://placehold.it/15/f03c15/000000?text=+)
 
-If you need to allow Pods of a particular application to connect
-to your application, you need to combine `podSelector` and `namespaceSelector`
-with an `AND` (intersection) operation. However, `spec.ingress.from[]` field of the
-NetworkPolicy object merges the result of  `podSelector` and `namespaceSelector`
-with an `OR` (union), making this impossible.
 
-For example, the following syntax is currently not possible as a single
-rule must have either a `namespaceSelector` or a `podSelector`, not both:
+Since Kubernetes v1.11, it is possible to combine `podSelector` and `namespaceSelector`
+with an `AND` (intersection) operation.
+
+## Example
+
+Start a `web` application:
+
+    kubectl run web --image=nginx \
+        --labels=app=web --expose --port 80
+
+Create a `other` namespace and label it:
+
+    kubectl create namespace other
+    kubectl label namespace/other team=operations
+
+The following manifest restricts traffic to only pods with label `type=monitoring` in namespaces labelled `team=operations`. Save it to `web-allow-all-ns-monitoring.yaml` and apply to the cluster:
 
 ```yaml
-ingress:
-  from:
-  - namespaceSelector: {}  # chooses all pods in all namespaces
-    podSelector:           # chooses pods with type=monitoring
-      matchLabels:
-        type: monitoring
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: web-allow-all-ns-monitoring
+  namespace: default
+spec:
+  podSelector:
+    matchLabels:
+      app: web
+  ingress:
+    - from:
+      - namespaceSelector:     # chooses all pods in namespaces labelled with team=operations
+          matchLabels:
+            team: operations  
+      - podSelector:           # chooses pods with type=monitoring
```

Comment:
I think this is wrong. The current example shows:

```yaml
    - from:
      - namespaceSelector:
          matchLabels:
            team: operations  
      - podSelector:
          matchLabels:
            type: monitoring
```

This will allow traffic from
- ALL pods in namespaces with `team: operations`
- ALL pods in the current namespace with label `type: monitoring`

And these two rules will be OR'ed not AND'ed (see `kubectl explain networkpolicy.spec.ingress.from`). It says:

> Items in this list are combined using a logical OR operation.

Therefore, is should be a single rule:

```yaml
    - from:
      - namespaceSelector:
          matchLabels:
            team: operations  
        podSelector:
          matchLabels:
            type: monitoring
```

A subtle difference, but it makes all the difference. :)